fairlearn/metrics/metrics_engine.py

```diff
@@ -8,7 +8,7 @@
 _MESSAGE_SIZE_MISMATCH = "Array {0} is not the same size as {1}"
 
 
-def metric_by_group(metric_function, y_true, y_pred, group_data, sample_weight=None):
+def metric_by_group(metric_function, y_true, y_pred, sensitive_features, sample_weight=None):
```

Comment:
Does `metric_by_group` need to change, now that we no longer have an argument `group_data` ?

Reply:
Also, in this case, `sensitive_features` has to be singular.

Reply:
Argh... i think that metric_by_group is actually okay to leave as is. I was thinking about leaving sensitive_features as a plural--since we are leaving it as a plural elsewhere even though we only support a single sensitive_feature everywhere. Thoughts?

Reply:
I had actually thought that was why we were using `group_data` here rather than `aux_data`  - because conceptually, we are just producing metrics by subgroups, without trying to treat them as sensitive, protected or whatever

Reply:
I see--very good point. I think I would go with group_membership here then. Apparently, group_data is confusing. Is that fine with you?

Reply:
`group_membership` is fine with me

Reply:
Updated to `group_membership` and pushed to branch